### PR TITLE
Organize methods inside utils.py into separate modules based on functionality

### PR DESCRIPTION
**Pull request by voith**
### What was wrong?
We are continuously adding helper methods to utils.py without following any structure. We should structure these into modules based on functionality. 

### How was it fixed?
organized methods inside utils.py into separate modules based on functionality

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->
picture credits: [peepeth.com/fhi](https://peepeth.com/fhi)

### CODE REVIEW

**Comment by voith** [comment]
@SamWilsn @lightclient Can you please review this?

File: src/ethereum/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+^^^^^^^^^^^^^^^^^
+
+..contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Utility functions used in this application.
```
Comment (SamWilsn) [inline src/ethereum/utils/__init__.py:12]:
```suggestion
Utility functions used in this specification.
```

File: src/ethereum/utils/address.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+..contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Address specific utility functions used in this application.
```
Comment (SamWilsn) [inline src/ethereum/utils/address.py:12]:
```suggestion
Address specific utility functions used in this specification.
```

File: src/ethereum/utils/address.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+"""
+from typing import Union
+
+from ethereum.base_types import U256, Uint
+from ethereum.frontier.eth_types import Address
```
Comment (SamWilsn) [inline src/ethereum/utils/address.py:17]: Something in the "parent" module probably shouldn't import something from inside the sub-module.

Case in point, when we introduce  the homestead fork, mypy will complain if we use homestead's `Address` type in `to_address`.

tl;dr `to_address` should be in frontier.

File: src/ethereum/utils/block.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    :local:
+
+Introduction
+------------
+
+Blocks specific utility functions used in this application.
+"""
+from typing import Any, Dict, Tuple
+
+from ethereum.frontier.eth_types import Block, Header
```
Comment (SamWilsn) [inline src/ethereum/utils/block.py:16]: Hm, I think the same suggestion applies here too. We can't really have `ethereum.frontier` types in `ethereum.utils`. Would an `ethereum.frontier.json` package make sense?

File: src/ethereum/utils/rlp.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+RLP specific utility functions used in this application.
+"""
+
+from ethereum import crypto
+from ethereum.crypto import Hash32
+from ethereum.frontier import rlp
+
+
+def rlp_hash(data: rlp.RLP) -> Hash32:
```
Comment (SamWilsn) [inline src/ethereum/utils/rlp.py:20]: Should probably just be in `ethereum.frontier.rlp` :/

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: src/ethereum/utils/hexadecimal.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    :local:
+
+Introduction
+------------
+
+Hexadecimal strings specific utility functions used in this specification.
+"""
+from ethereum.base_types import U256, Bytes, Bytes8, Bytes32, Uint
+from ethereum.crypto import Hash32
+from ethereum.frontier.eth_types import Address, Bloom, Root
```
Comment (SamWilsn) [inline src/ethereum/utils/hexadecimal.py:16]: Whoops, I missed one. I'll put up a PR myself to clean the rest of these up.

**Review by SamWilsn — APPROVED** [review]
